File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/stack.java
/**
 * Java Implementation: Stack
 * Invariant: Push/Pop/Peek O(1), Space O(N)
 */
package seed.dsa.core;

import java.util.*;

public class Stack<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/c/stack.c
/*
 * C Implementation: Stack
 * Invariant: Push/Pop/Peek O(1), Space O(N)
 */
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* data;
    int size;
    int capacity;
} stack_t;

// TODO: Complete the allocation and operations
void init_stack(stack_t* item, int capacity) {
    item->data = (int*)malloc(capacity * sizeof(int));
    item->size = 0;
    item->capacity = capacity;
}
